src/api/rolesApi.tsx

```diff
@@ -105,14 +119,36 @@ export const useAddRole = () => {
 
 export const useDeleteRole = () => {
   const queryClient = useQueryClient();
+  const roles = useRoles();
   const { dispatchToast } = useToastController("toaster");
   return useMutation(
     ["deleteRole"],
     async (id: number) => {
-      return spWebContext.web.lists
+      let error = undefined;
+      const thisRole = roles.data?.find((role) => (role.Id = id));
+
+      const result = await spWebContext.web.lists
         .getByTitle("roles")
         .items.getById(id)
-        .recycle();
+        .recycle()
+        .catch((err) => (error = Promise.reject(new Error(err))));
+
+      if (!error) {
+        const usersRoles = roles.data?.filter((role) => {
+          role.user.Id === thisRole?.user.Id;
+        });
```

Comment:
This always returns an empty array -- remove the curly braces so the filter will work as desired

```suggestion
        const usersRoles = roles.data?.filter(
          (role) => role.user.Id === thisRole?.user.Id
        );
```